pkg/testutils/copyfile.go

```diff
@@ -28,3 +30,38 @@ func CopyFile(toPath, fromPath string, perm os.FileMode) error {
 	}
 	return err
 }
+
+func CopyFileToTmp(fname string) (string, error) {
+	bname := path.Base(fname)
+	pattern := fmt.Sprintf("%s.*", bname)
+	outf, err := os.CreateTemp("", pattern)
+	if err != nil {
+		return "", err
+	}
+	defer outf.Close()
+
+	inf, err := os.Open(fname)
+	if err != nil {
+		os.Remove(outf.Name())
```

Comment:
why not using `defer` instead of having multiple `os.Remove(outf.Name())` at various return? Since they are done in reverse order you could:
```go
defer os.Remove(outf.Name())
defer outf.Close()
```

Reply:
Dunno, for error handling I prefer using `os.Remove()` and t.Cleanup() for cleanup. I think the function is small enough that it does not matter.